src/pages/restaurant/components/dishItems/DishItems.js

```diff
@@ -1,32 +1,93 @@
 // react
-import React, { useMemo } from "react";
+import React, { useMemo, useState } from "react";
 
 //prop-types
 import PropTypes from "prop-types";
 
 //lodash
 import _keys from "lodash/keys";
+import _isEmpty from "lodash/isEmpty";
+
+// helpers
+import {
+  filterItemIfQuantityIsZero,
+  findItemById,
+  updateItemQuantity,
+} from "./helpers/dishItems.helpers";
+import { saveCartItems } from "../../../../utility/localStorage";
 
 // components
 import Sidebar from "../../../../commonComponents/sidebar";
 import Menu from "./components/menu";
+import EmptyCart from "../../../../commonComponents/cart/components/emptyCart";
+import Cart from "../../../../commonComponents/cart";
 
 // css
 import "./dishItems.css";
 
-const DishItems = (props) => {
-  const { dishItems } = props;
+const DishItems = ({ dishItems }) => {
+  const [cart, setCart] = useState({
```

Comment:
since you are using only items array inside this object
you can directly use an array as a state which will easy to handle

Reply:
I was thinking to keep track of cart ID because in real scenario it will be used by multiple users so associating ID with each cart is important, right now I've removed it from the implementation

Reply:
okay